src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1737,6 +1744,52 @@ private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax
 
             var checkedVariables = checkedVariablesBuilder.ToImmutableAndFree();
 
+            // tuple literal such as `(1, 2)`, `(null, null)`, `(x.P, y.M())`
+            if (boundRHS.Kind == BoundKind.TupleLiteral || ((object)boundRHS.Type != null && boundRHS.Type.IsTupleType))
+            {
+                return BindDeconstructWithTuple(node, diagnostics, boundRHS, checkedVariables);
+            }
+
+            // expression without type such as `null`
+            if (boundRHS.Type == null)
+            {
+                Error(diagnostics, ErrorCode.ERR_DeconstructRequiresExpression, node);
+                return BadExpression(node, new[] { boundRHS });
+            }
+
+            return BindDeconstructWithDeconstruct(node, diagnostics, boundRHS, checkedVariables);
+        }
+
+        private BoundExpression BindDeconstructWithTuple(AssignmentExpressionSyntax node, DiagnosticBag diagnostics, BoundExpression boundRHS, ImmutableArray<BoundExpression> checkedVariables)
+        {
+            // patch up the tuple literal so that it has a type, based on the LHS information
+            var lhsTypes = checkedVariables.SelectAsArray(v => v.Type);
+            TypeSymbol lhsAsTuple = TupleTypeSymbol.Create(locationOpt: null, elementTypes: lhsTypes, elementLocations: default(ImmutableArray<Location>), elementNames: default(ImmutableArray<string>), compilation: Compilation, diagnostics: diagnostics);
```

Comment:
I think locations should not be set here because this is not a declaration location for the tuple type.

---
In reply to: [64080194](https://github.com/dotnet/roslyn/pull/11457#discussion_r64080194) [](ancestors = 64080194)